File: leetcode/1005.maximize-sum-of-array-after-k-negations/cpp/largestsumafterknegations.cpp
#include <vector>
#include <algorithm>

class Solution
{
public:
    int largestSumAfterKNegations(std::vector<int> &nums, int k)
    {
        std::sort(nums.begin(), nums.end(), [](int a, int b)
                  { return std::abs(b) < std::abs(a); });

        int res = 0;

        for (auto v : nums)
        {
            if (v < 0 && k > 0)
            {
                res += -v;
                k--;
                continue;
            }
            res += v;
        }

        if (k % 2 == 1)
            res -= 2 * std::abs(nums.back());

        return res;
    }
};

File: leetcode/1005.maximize-sum-of-array-after-k-negations/ts/largestsumafterknegations.ts
const largestSumAfterKNegations = (nums: number[], k: number): number => {
    nums.sort((a, b) => Math.abs(b) - Math.abs(a))
    let res = 0

    for (const v of nums) {
        if (v < 0 && k > 0) {
            res += Math.abs(v)
            k--
            continue
        }
        res += v
    }

    if (k % 2 === 1) res -= 2 * Math.abs(nums[nums.length - 1])

    return res
}
